csrc/scheduler/registry.cpp

```diff
@@ -1296,7 +1296,12 @@ class NoOpScheduler : public SchedulerEntry {
         auto all_nonzero = std::none_of(
             concrete_dimension.begin(),
             concrete_dimension.end(),
-            [](IterDomain* id) { return id->extent()->isZeroInt(); });
+            [](IterDomain* id) {
+              // Purely symbolic extents _might_ be zero depending on inputs.
+              // These will need to be checked at runtime. So we should only
+              // reject if the extent is known to be non-zero at compile time.
+              return !id->extent()->isConstInt() || id->extent()->isZeroInt();
```

Comment:
Shouldn't this check be also done at the run time? The compile-time check only makes sure there may be a size-0 non-reduction domain, but it's not guaranteed.

Reply:
Yes it needs a runtime check too.

Reply:
I wonder if we should treat zero size as a special type like we do with one size so if a size changes in/out of zero we always recompile. cc @jjsjann123